src/vcf.rs

```diff
@@ -515,28 +550,52 @@ impl VCFFile {
             }
         }
 
-        calls.push((indel_start as usize, _indel_type, indel_str));
+        calls.push((indel_start, _indel_type.clone(), indel_str));
 
-        for i in 0.._x.len() {
+        if _indel_type == AltType::INS {
+            // Return to vector index now the call has been constructed
+            indel_start += 1;
+        }
+        let mut indel_left = length;
+        let mut offset = 0;
+        for i in 0.._y.len() {
+            if i as i64 == indel_start {
+                // Pad where the indel is to ensure we get all calls
+                if indel_left > 0 {
+                    indel_left -= 1;
+                    indel_start += 1;
+                    offset += 1;
+                }
+                continue;
+            }
             let r = _y.chars().nth(i).unwrap();
-            let a = _x.chars().nth(i).unwrap();
+            let a = _x.chars().nth(i - offset).unwrap();
             if r != 'N' && a != 'N' && r != a {
-                calls.push((i, AltType::SNP, a.to_string()));
+                if _indel_type == AltType::DEL {
+                    calls.push((i as i64, AltType::SNP, a.to_string()));
+                } else {
+                    calls.push((i as i64, AltType::SNP, r.to_string()));
+                }
             }
         }
-        return calls;
+
+        // I hate implict returns, but appease clippy
+        calls
     }
 }
 
 /// Calculate the distance between two strings ignoring N characters
 ///
 /// # Arguments
-/// - `reference`: &String - Reference sequence
-/// - `alternate`: &String - Alternate sequence
+/// - `reference`: &str - Reference sequence
+/// - `alternate`: &str - Alternate sequence
 ///
 /// # Returns
 /// SNP distance between the two strings
-fn snp_dist(reference: &String, alternate: &String) -> i64 {
+fn snp_dist(reference: &str, alternate: &str) -> i64 {
+    if reference.len() != alternate.len() {
+        panic!("Reference and alternate strings are not the same length!");
+    }
     let mut dist = 0;
     for i in 0..reference.len() {
```

Comment:
I'm not sure you'll want to take the length of a str like that (https://stackoverflow.com/questions/46290655/does-rusts-string-have-a-method-that-returns-the-number-of-characters-rather-th)
I imagine it's fine here as we know they are normal utf8 characters.
I would consider rewriting this with a zip of iterators:
```
fn count_differences(s1: &str, s2: &str) -> usize {
    s1.chars().zip(s2.chars()).filter(|(c1, c2)| c1 != c2 && c1 != 'N' && c2 != 'N').count()
}
```

Reply:
Yeah I saw that, but honestly if a VCF has non UTF-8 characters in it there's going to be larger issues than just that so I think simplistic code is better here